src/common/providers/database.ts

```diff
@@ -128,7 +128,7 @@ export class DataSnapshot implements database.DataSnapshot {
     let source = this._data;
     if (parts.length) {
       for (const part of parts) {
-        if (source[part] === undefined) {
+        if (typeof source === "undefined" || source === null) {
```

Comment:
As well as moving this "up" to the parent object I've also changed the check from `=== undefined` to a `typeof` so it matches the other checks elsewhere:

https://github.com/firebase/firebase-functions/blob/5f7b331abd5a2a38a4e84de1e51411ea5919c0c3/src/common/providers/database.ts#L173

https://github.com/firebase/firebase-functions/blob/5f7b331abd5a2a38a4e84de1e51411ea5919c0c3/src/common/providers/database.ts#L278